### ISSUE TITLE
How to use @mention

### ISSUE THREAD

**Issue by fc-ds** [atc0005/go-teams-notify#127]
Hello,

I'm trying to mention someone in the cards like I do manually with @user .

Is there a way to do that with this lib ?

**Comment by ghokun** [comment]
You can mention someone with following:

```bash
curl -X POST -H "Content-type: application/json" -d '{
    "type": "message", 
    "text": "Hey <at>Some User</at> check out this message",
    "entities": [
        {
            "type":"mention",
            "mentioned":{
                "id":"some.user@company.com",
                "name":"Some User"
            },
            "text": "<at>Some User</at>"
        }
    ]
}' <webhook_url>
```

**Comment by atc0005** [comment]
@ghokun thanks for sharing!

**Comment by atc0005** [comment]
@fc-ds Thanks to the feedback from @ghokun, we now know that this (much requested) feature is available in Microsoft Teams.

I'll first prototype the support in a small project that I maintain which uses this library, then land the applicable changes here. No estimates for when that will be, but I'll be sure to let you know when it does.

Thanks for opening this issue and your interest.

**Comment by atc0005** [comment]
> @ghokun: You can mention someone with following:
> 
> ```shell
> curl -X POST -H "Content-type: application/json" -d '{
>     "type": "message", 
>     "text": "Hey <at>Some User</at> check out this message",
>     "entities": [
>         {
>             "type":"mention",
>             "mentioned":{
>                 "id":"some.user@company.com",
>                 "name":"Some User"
>             },
>             "text": "<at>Some User</at>"
>         }
>     ]
> }' <webhook_url>
> ```

As previously noted, this example works as intended. However, I'm having some trouble finding a good reference for the JSON format.

This doesn't appear to be the [legacy MessageCard format](https://docs.microsoft.com/en-us/outlook/actionable-messages/message-card-reference) or the [Adaptive Card format](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-format?tabs=adaptive-md%2Cconnector-html#mention-support-within-adaptive-cards), but potentially the format noted here:

https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/channel-and-group-conversations?tabs=json#add-mentions-to-your-messages

If this library implements the Adaptive Card format, the desired mention support would be provided, but the JSON snippet you provided appears to be much simpler. If this library implements support for that format, the desired functionality can be added sooner.

I expect that I'm overlooking it, but if you happen to have a reference to share that would be appreciated.

**Comment by ghokun** [comment]
I was looking for a way to mention user names when a pull request is opened on gitlab through gitlab-teams integration. I was not successful but saw this repository. I thinks this message format is for bot api but works in webhooks regardless.

There is a website for designing adaptive cards : https://adaptivecards.io/designer/
I spent some time there but could not get mentions to work with adaptive cards.

**Comment by atc0005** [comment]
@fc-ds 

Initial support available here:

https://github.com/atc0005/go-teams-notify/releases/tag/v2.7.0-alpha.1

Example here:

https://github.com/atc0005/go-teams-notify/blob/2b9b7215a7ca1c7564eba47609611c3fe4d34997/examples/user-mention/main.go#L35-L58

Feedback is welcome.

This is a pretty limited implementation; a more complete implementation is planned for a future release once `Adaptive Card` support is added. This upcoming release is intended to lay the foundation for that future support.

**Comment by atc0005** [comment]
> @fc-ds
> 
> Initial support available here:
> 
> https://github.com/atc0005/go-teams-notify/releases/tag/v2.7.0-alpha.1
> 
> Example here:
> 
> https://github.com/atc0005/go-teams-notify/blob/2b9b7215a7ca1c7564eba47609611c3fe4d34997/examples/user-mention/main.go#L35-L58
> 
> Feedback is welcome.
> 
> This is a pretty limited implementation; a more complete implementation is planned for a future release once `Adaptive Card` support is added. This upcoming release is intended to lay the foundation for that future support.

Initial `Adaptive Card` support has been added as of this release:

https://github.com/atc0005/go-teams-notify/releases/tag/v2.7.0-alpha.2

Feedback is very much encouraged so issues can be found/resolved before the stable v2.7.0 release (date TBD). Discussion thread here: https://github.com/atc0005/go-teams-notify/discussions/163

The `botapi` package intended as temporary support for user mentions has been removed as the generated payloads have been found to be intermittently rejected by the Teams API (atc0005/send2teams#225).

Examples for the new package can be found here:

https://github.com/atc0005/go-teams-notify/tree/master/examples/adaptivecard

An equivalent example for what I provided previously:

https://github.com/atc0005/go-teams-notify/blob/b24de89b00d25156a9e2f6fb8796b84a99bf48e2/examples/adaptivecard/user-mention-single/main.go#L37-L65

Thanks for your interest in this project.

**Comment by atc0005** [comment]
No negative feedback has been received for the v2.7.0-alpha.2 release thus far.

I'm considering this resolved for the v2.7.0 release.
